config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
```

Comment:
> a function similar to RecoverCluster() that doesn't take a new configuration but rolls up the logs and takes a snapshot, writing that out in the latest version. 

That implicitly commits everything, so we can't do that safely.

> it could maybe be in NewRaft() as something we do automatically for folks

Maybe after an upgrade once your commit index has reached ???, you force a snapshot. Filling in the ??? is tricky, hmm.